cvat/apps/quality_control/quality_reports.py

```diff
@@ -1994,27 +1994,14 @@ def generate_report(self) -> ComparisonReport:
                 annotations = deepcopy(frame_result.annotations)
             else:
                 annotations.accumulate(frame_result.annotations)
-            confusion_matrices.append(frame_result.annotations.confusion_matrix.rows)
+            confusion_matrix += frame_result.annotations.confusion_matrix.rows
 
             if annotation_components is None:
                 annotation_components = deepcopy(frame_result.annotation_components)
             else:
                 annotation_components.accumulate(frame_result.annotation_components)
             mean_ious.append(frame_result.annotation_components.shape.mean_iou)
 
-        confusion_matrix_labels = {
-            i: label.name
-            for i, label in enumerate(self._gt_dataset.categories()[dm.AnnotationType.label])
-            if not label.parent
-        }
-        confusion_matrix_labels[None] = "unmatched"
-        confusion_matrix_labels_rmap = {k: i for i, k in enumerate(confusion_matrix_labels.keys())}
-        if confusion_matrices:
-            confusion_matrix = np.sum(confusion_matrices, axis=0)
```

Comment:
Why the matrix accumulation code was removed?

Reply:
It's not removed, it's just done in the loop above now.

Reply:
It's less efficient in the new variant, I suppose. There can be quite a few frames in the task, and the old variant summed them with numpy.

Reply:
Okay, I did a little benchmark, and at matrix sizes where the time taken non-negligible the new variant is faster. 🤷

Reply:
Looks strange, but if so, okay. Maybe it's better to just include this in the `accumulate()` then. There is also a parallel [PR](https://github.com/opencv/cvat/pull/7604) that touches this topic.